src/Session/SessionManager.cc

```diff
@@ -185,22 +187,30 @@ void SessionManager::OnMessage(WSType* ws, std::string_view sv_message, uWS::OpC
         session->UpdateLastMessageTimestamp();
 
         EventHeader head = Message::GetEventHeader(sv_message);
-
         CARTA::EventType event_type = head.GetType();
+
+        if (!CARTA::EventType_IsValid(event_type)) {
+            spdlog::error("Bad event type: {}", event_type);
+            session->SendLogEvent("Bad event type: " + event_type, {"event"}, CARTA::ErrorSeverity::ERROR);
```

Comment:
Please create a string variable for the error message (using `fmt::format`) and reuse it for both the local log and the event.

Normally you would avoid doing this kind of interpolation before using the log function (because it interpolates on demand depending on the log level), but in this case avoiding repetition is more important, and typically the log level will include errors.

(This also applies to the other two places.)